JavaScript/JavaScriptSDK/Util.ts

```diff
@@ -148,5 +148,14 @@
                 && columnNumber == 0
                 && error == null;
         }
+
+        /**
+        * Returns string representation of an object suitable for diagnostics logging.
+        */
+        public static dump(object: any): string {
+            var objectTypeDump: string = Object.prototype.toString.call(object);
+            var propertyValueDump: string = JSON.stringify(object);
```

Comment:
I think this pseudo code makes sense - if error is big it gets cut off, if there's any problem (like circ ref) we'll report it as well and then will be able to analyze it later - it perfectly follows our current strategy data-driven decisions for fixing/improving the code.

try { 
  res = stringify(...) 

  size = size(res)
  if (tooBig(size)) {
    cutOff = true;
    res = "[object too big(" + size + ")]:" + shink(res);
  }

  return res;
} 
catch (e) { 
  InternalLogging.Critical("couldn't stringify"); 
}